File: Sorting/Counting Sort/C++/count_sort.cpp
#include<bits/stdc++.h>
using namespace std;
#define RANGE 1000

void countSort(int arr[], int n) 
{ 
	int output[n]={0}; 
	int count[RANGE + 1]={0}; 
	 

	for (int i = 0; i<n ; i++) 
	{
		count[arr[i]]++; 
	}

	for (int i = 1; i <= RANGE; i++) 
	{
		count[i] += count[i - 1]; 
	}

	for (int i = 0; i<n ; i++) 
	{ 
		output[count[arr[i]] - 1] = arr[i]; 
		count[arr[i]]--; 
	} 

	
	for (int i = 0; i<n; ++i) 
		cout<<output[i]<<" ";
	cout<<endl;
} 


int main() 
{ 
	int arr[] = {1, 4, 1, 2, 7, 5, 2}; 
    int n = sizeof(arr)/sizeof(arr[0]);
	countSort(arr, n); 

	
} 

File: Sorting/Counting Sort/Python/count_sort.py
def countSort(arr, n):
	RANGE = 1000
	count = [0 for i in range(RANGE)]
	output = [0 for i in range(n)]
	for i in range(n):
		count[arr[i]]+= 1
	for i in range(1,RANGE):
		count[i]+= count[i-1]
	for i in range(n):
		output[count[arr[i]]-1] =arr[i]
		count[arr[i]]-=1
	print(output)

arr = [1, 4, 1, 2, 7, 5, 2]
n = len(arr)
countSort(arr,n)